src/main/java/com/scalar/cassy/util/RemoteFileSystemConnection.java

```diff
@@ -0,0 +1,43 @@
+package com.scalar.cassy.util;
+
+import com.google.common.base.Preconditions;
+import com.palantir.giraffe.host.Host;
+import com.palantir.giraffe.host.HostControlSystem;
+import com.palantir.giraffe.ssh.PublicKeySshCredential;
+import com.palantir.giraffe.ssh.SshCredential;
+import com.palantir.giraffe.ssh.SshHostAccessor;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Paths;
+
+public class RemoteFileSystemConnection {
+  private final HostControlSystem hostControlSystem;
+  private final String storagePath;
+
+  public RemoteFileSystemConnection(URI storageBaseURI) throws IOException {
+    SshHostAccessor sshHostAccessor = initSshHostAccessor(storageBaseURI);
+    this.hostControlSystem = sshHostAccessor.open();
+    this.storagePath = storageBaseURI.getPath();
+  }
+
+  private SshHostAccessor initSshHostAccessor(URI storageBaseURI) throws IOException {
+    Preconditions.checkArgument(storageBaseURI.getScheme().equals("ssh"));
+    String userName =
+        storageBaseURI.getUserInfo() != null
+            ? storageBaseURI.getUserInfo()
+            : System.getProperty("user.name");
+    SshCredential credential =
+        PublicKeySshCredential.fromFile(
+            userName, Paths.get(System.getProperty("user.home"), ".ssh/id_rsa"));
```

Comment:
Instead of hard-code this, we can assume that there is a ssh.config that configures required settings to access the remote server. What do you think?

Reply:
Indeed, it would be nice if Giraffe could infer directly the ssh authentication settings for a given host but it does not seem possible.
Looking at the [SshHostAccessor](https://palantir.github.io/giraffe/docs/0.10.1/api/com/palantir/giraffe/ssh/SshHostAccessor.html) which is the Giraffe object used to instantiate ssh connection. The only methods available require to provide either the public ssh key or a password. A third way is to use the kerberos protocol but I am not familiar with it.